src/pages/ListScholarships.jsx

```diff
@@ -58,6 +79,21 @@ function ListScholarships() {
     pruneQueryParam(qParams.MAX_AMOUNT);
   }
 
+  /**
+   * prunes invalid grade values not respresented by GradeLevel enum
+   */
+  if (grades !== undefined) {
+    if (Array.isArray(grades) && grades.length > 0) {
+      const prunedInvalid = [...new Set(grades)].filter((g) =>
+        GradeLevel.keys().includes(g)
+      );
+
+      if (prunedInvalid.length !== grades.length) {
+        replaceQueryParam(qParams.GRADES, prunedInvalid);
+      }
+    } else pruneQueryParam(qParams.GRADES);
+  }
```

Comment:
tested against the following scenarios:
`grades`
`grades[]`
`grades[]=`
`grades[]=1,2,8`

Reply:
I'm guessing this won't prune though when you do `grades[]=1,2,3` when all the values are invalid.

Reply:
it does, first it replaces it with `[]` and then that gets pruned 😃